### ISSUE TITLE
Adds Microsoft Cognitive Toolkit Library

### ISSUE THREAD

**Issue by AlexCatarino** [QuantConnect/Lean#4125]
#### Expected Behavior
Support [Microsoft Cognitive Toolkit Library](https://docs.microsoft.com/en-us/cognitive-toolkit/):
> The Microsoft Cognitive Toolkit (CNTK) is an open-source toolkit for commercial-grade distributed deep learning. It describes neural networks as a series of computational steps via a directed graph. CNTK allows the user to easily realize and combine popular model types such as feed-forward DNNs, convolutional neural networks (CNNs) and recurrent neural networks (RNNs/LSTMs). CNTK implements stochastic gradient descent (SGD, error backpropagation) learning with automatic differentiation and parallelization across multiple GPUs and servers.

This library has a [C#](https://docs.microsoft.com/en-us/cognitive-toolkit/nuget-package) and [Python](https://docs.microsoft.com/en-us/cognitive-toolkit/Setup-Linux-Python?tabs=cntkpy27) version.

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
